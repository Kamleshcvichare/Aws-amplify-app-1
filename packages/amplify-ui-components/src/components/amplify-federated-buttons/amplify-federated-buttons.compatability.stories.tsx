/* @jsx createElement */

import { Auth } from '@aws-amplify/auth';
import { FederatedButtons } from 'aws-amplify-react';
import { createElement, StrictMode } from 'react';
import { render } from 'react-dom';

// Mock to match `aws-amplify-react/__tests__/Auth/FederatedSignIn-test.tsx`
Auth.configure = () => ({});

const noop = () => {};

export default {
  title: 'amplify-federated-buttons/FederatedButtons',
  decorators: [
    story => {
      const node = document.createElement('main');
      render(<StrictMode>{story()}</StrictMode>, node);
      return node;
    },
  ],
};

export const renderWithCorrectAuthState = () => (
  <FederatedButtons
    authState="signIn"
    federated={{
      google_client_id: 'google_client_id',
      facebook_app_id: 'facebook_app_id',
    }}
    onStateChange={noop}
  />
);

export const renderWithCorrectAuthStateAndOnlyFacebookId = () => (
  <FederatedButtons
    authState="signIn"
    federated={{
      facebook_app_id: 'facebook_app_id',
    }}
    onStateChange={noop}
  />
);

export const renderWithCorrectAuthStateAndOnlyGoogleId = () => (
  <FederatedButtons
    authState="signIn"
    federated={{
      google_client_id: 'google_client_id',
    }}
    onStateChange={noop}
  />
);

export const renderNothingWithIncorrectAuthState = () => (
  <FederatedButtons authState="signedIn" federated={{}} onStateChange={noop} />
);

export const renderNothingWithNoFederatedProp = () => (
  <FederatedButtons authState="signedIn" federated={undefined} onStateChange={noop} />
);
