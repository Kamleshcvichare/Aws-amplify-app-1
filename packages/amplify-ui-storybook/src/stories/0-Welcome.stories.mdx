import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import { AmplifyAuthenticator } from '@aws-amplify/ui-react';

<Meta title="Welcome" />

# Amplify UI Components (Preview)

[Amplify UI Components (Preview)](#) is an open-source UI component library that encapsulates cloud-connected workflows inside of framework-agnostic UI components.

## Frameworks

| Framework          | Package                                                                                  | Version                                                                                                                                        |                    READMEs                     | Quick Start                         |
| ------------------ | ---------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- | :--------------------------------------------: | ----------------------------------- |
| **React**          | [`@aws-amplify/ui-react`](https://www.npmjs.com/package/@aws-amplify/ui-react)           | [![version](https://img.shields.io/npm/v/@aws-amplify/ui-react/ui-preview.svg)](https://www.npmjs.com/package/@aws-amplify/ui-react)           |  [`README.md`](../amplify-ui-react/README.md)  | [`React`](#react)                   |
| **Angular**        | [`@aws-amplify/ui-angular`](https://www.npmjs.com/package/@aws-amplify/ui-angular)       | [![version](https://img.shields.io/npm/v/@aws-amplify/ui-angular/ui-preview.svg)](https://www.npmjs.com/package/@aws-amplify/ui-angular)       | [`README.md`](../amplify-ui-angular/README.md) | [`Angular`](#angular)               |
| **Vue**            | [`@aws-amplify/ui-vue`](https://www.npmjs.com/package/@aws-amplify/ui-vue)               | [![version](https://img.shields.io/npm/v/@aws-amplify/ui-vue/ui-preview.svg)](https://www.npmjs.com/package/@aws-amplify/ui-vue)               |   [`README.md`](../amplify-ui-vue/README.md)   | [`Vue`](#vue)                       |
| **Web Components** | [`@aws-amplify/ui-components`](https://www.npmjs.com/package/@aws-amplify/ui-components) | [![version](https://img.shields.io/npm/v/@aws-amplify/ui-components/ui-preview.svg)](https://www.npmjs.com/package/@aws-amplify/ui-components) |            [`README.md`](README.md)            | [`Web Components`](#web-components) |

## Installation

<details>
	<summary>React</summary>

```
yarn add aws-amplify@ui-preview @aws-amplify/ui-react@ui-preview
```


</details>
