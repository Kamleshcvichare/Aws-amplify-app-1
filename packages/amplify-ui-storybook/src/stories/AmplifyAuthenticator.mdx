import { Props, Story } from '@storybook/addon-docs/blocks';

# AmplifyAuthenticator

The Authenticator is a drop-in UI component that provides:

- User Sign in
- User Sign up
- User Sign out
- Forgot Password
- Federated authentication
- MFA (Multi-Factor Authentication) e.g. SMS, Email, and TOTP (Temporary One Time Password)
- Confirm MFA Codeâ€™s and Provide QR codes for TOTP

<Props of="amplify-authenticator" />

## Live Demo

<Story id="amplifyauthenticator--basic" />

## Social Providers

<Story id="amplifyauthenticator--federated-identity-providers" />

## Initial State
