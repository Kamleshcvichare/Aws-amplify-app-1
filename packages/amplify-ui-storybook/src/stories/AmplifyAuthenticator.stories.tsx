import { AmplifyAuthenticator } from '@aws-amplify/ui-react';
import { withKnobs, select, text } from '@storybook/addon-knobs';
import React from 'react';

import page from './AmplifyAuthenticator.mdx';

export default {
	title: 'AmplifyAuthenticator',
	component: 'amplify-authenticator',
	decorators: [withKnobs],
	parameters: { docs: { page } },
};

export const basic = () => <AmplifyAuthenticator />;

export const federatedIdentityProviders = () => (
	<AmplifyAuthenticator
		federated={{
			amazonClientId: text('Amazon client ID', 'amazon_client_id'),
			auth0Config: {
				clientID: text('Auth0 client ID', 'auth0_client_id'),
				domain: text('Auth0 account domain', 'example.auth0.com'),
				redirectUri: 'http://localhost:3000/',
				responseType: 'token id_token',
			},
			facebookAppId: text('Facebook app ID', 'facebook_app_id'),
			googleClientId: text('Google client ID', 'google_client_id'),
			oauthConfig: {},
		}}
	/>
);

export const initialState = () => {
	const initialAuthState = select(
		'Initial Auth State',
		['signin', 'signup'],
		'signup'
	);

	return (
		<AmplifyAuthenticator
			// Dispose when initialAuthState changes
			key={initialAuthState}
			// AuthState isn't exported
			initialAuthState={initialAuthState as any}
		/>
	);
};
