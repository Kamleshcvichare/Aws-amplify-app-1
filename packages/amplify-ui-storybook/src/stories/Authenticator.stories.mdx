import { Meta, Props, Story, Preview } from '@storybook/addon-docs/blocks';
import { AmplifyAuthenticator } from '@aws-amplify/ui-react';
import { withKnobs, text } from '@storybook/addon-knobs';

<Meta title="Authenticator" decorators={[withKnobs]} />

# Authenticator

The Authenticator is a drop-in UI component that provides:

- User Sign in
- User Sign up
- User Sign out
- Forgot Password
- Federated authentication
- MFA (Multi-Factor Authentication) e.g. SMS, Email, and TOTP (Temporary One Time Password)
- Confirm MFA Codeâ€™s and Provide QR codes for TOTP

## Live Demo

<Preview>
	<Story name="Demo">
		<AmplifyAuthenticator />
	</Story>
</Preview>

---

<Props of="amplify-authenticator" />

Use `AmplifyAuthenticator` to render whatever

<Preview>
	<Story name="with Federated Signin">
		<AmplifyAuthenticator
			federated={{
				amazonClientId: text('Amazon client ID', 'amazon_client_id'),
				auth0Config: {
					clientID: text('Auth0 client ID', 'auth0_client_id'),
					domain: text('Auth0 account domain', 'example.auth0.com'),
					redirectUri: 'http://localhost:3000/',
					responseType: 'token id_token',
				},
				facebookAppId: text('Facebook app ID', 'facebook_app_id'),
				googleClientId: text('Google client ID', 'google_client_id'),
				oauthConfig: {},
			}}
		/>
	</Story>
</Preview>
