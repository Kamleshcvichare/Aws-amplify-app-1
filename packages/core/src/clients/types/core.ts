/**
 * General None HTTP-specific request interface
 */
export interface Request {
	url: URL;
	body?: unknown;
}

export interface Response {
	body: unknown;
}

export interface TransferHandler<
	Input extends Request,
	Output extends Response,
	TransferOptions
> {
	(request: Input, options: TransferOptions): Promise<Output>;
}

/**
 * A slimmed down version of the AWS SDK v3 middleware handler, only handling instantiated requests
 */
export type MiddlewareHandler<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (request: Input, options: MiddlewareOptions) => Promise<Output>;

export type MiddlewareContext = Record<string, unknown>;

/**
 * A slimmed down version of the AWS SDK v3 middleware, only handling tasks after Serde.
 */
export type Middleware<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (
	next: MiddlewareHandler<Input, Output, MiddlewareOptions>,
	context: MiddlewareContext
) => MiddlewareHandler<Input, Output, MiddlewareOptions>;
