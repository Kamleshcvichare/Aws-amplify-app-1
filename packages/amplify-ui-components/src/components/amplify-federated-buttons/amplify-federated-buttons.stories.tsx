import { AmplifyFederatedButtons } from './amplify-federated-buttons';

const handleStateChange = (...args) => {
  console.log('handleStateChange', ...args);
};

export default {
  title: 'amplify-federated-buttons',
};

export const renderWithCorrectAuthState = () => {
  const element = new AmplifyFederatedButtons();

  element.authState = 'signIn';
  element.federated = {
    google_client_id: 'google_client_id',
    facebook_app_id: 'facebook_app_id',
  };

  element.handleAuthStateChange = handleStateChange;

  return element;
};

export const renderWithCorrectAuthStateAndOnlyFacebookId = () => {
  const element = new AmplifyFederatedButtons();

  element.authState = 'signIn';
  element.federated = {
    facebook_app_id: 'facebook_app_id',
  };
  element.handleAuthStateChange = handleStateChange;

  return element;
};

export const renderWithCorrectAuthStateAndOnlyGoogleId = () => {
  const element = new AmplifyFederatedButtons();

  element.authState = 'signIn';
  element.federated = {
    google_client_id: 'google_client_id',
  };
  element.handleAuthStateChange = handleStateChange;

  return element;
};

export const renderNothingWithIncorrectAuthState = () => {
  const element = new AmplifyFederatedButtons();

  // @ts-ignore intentionally setting invalid state
  element.authState = 'someInvalidState';
  element.federated = {};
  element.handleAuthStateChange = handleStateChange;

  return element;
};

export const renderNothingWithNoFederatedProp = () => {
  const element = new AmplifyFederatedButtons();

  // @ts-ignore intentionally setting invalid state
  element.authState = 'someInvalidState';
  element.federated = undefined;
  element.handleAuthStateChange = handleStateChange;

  return element;
};
