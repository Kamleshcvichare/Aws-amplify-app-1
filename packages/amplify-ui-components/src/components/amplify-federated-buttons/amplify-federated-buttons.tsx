import { Auth } from '@aws-amplify/auth';
import { isEmpty, I18n } from '@aws-amplify/core';
import { Component, h, Prop } from '@stencil/core';

@Component({ tag: 'amplify-federated-buttons' })
export class AmplifyFederatedButtons {
  /** The current authentication state. */
  @Prop() authState: 'signIn' | 'signedOut' | 'signedUp' = 'signIn';
  /** Federated credentials & configuration. */
  @Prop() federated: any = {};

  componentWillLoad() {
    if (!Auth || typeof Auth.configure !== 'function') {
      throw new Error('No Auth module found, please ensure @aws-amplify/auth is imported');
    }

    const { oauth = {} } = Auth.configure({});

    // backward compatibility
    if (oauth['domain']) {
      this.federated.oauth_config = { ...this.federated.oauth_config, ...oauth };
    } else if (oauth['awsCognito']) {
      this.federated.oauth_config = { ...this.federated.oauth_config, ...oauth['awsCognito'] };
    }

    if (oauth['auth0']) {
      this.federated.auth0 = { ...this.federated.auth0, ...oauth['auth0'] };
    }
  }

  render() {
    if (!['signIn', 'signedOut', 'signedUp'].includes(this.authState)) {
      return null;
    }

    if (isEmpty(this.federated)) {
      return null;
    }

    const { amazon_client_id, auth0, facebook_app_id, google_client_id, oauth_config } = this.federated;

    return (
      <div>
        {google_client_id && (
          <div>
            <amplify-google-button google_client_id={google_client_id} />
          </div>
        )}

        {facebook_app_id && (
          <div>
            <amplify-facebook-button facebook_app_id={facebook_app_id} />
          </div>
        )}

        {amazon_client_id && (
          <div>
            <amplify-amazon-button amazon_client_id={amazon_client_id} />
          </div>
        )}

        {oauth_config && (
          <div>
            <amplify-oauth-button oauth_config={oauth_config} />
          </div>
        )}

        {auth0 && (
          <div>
            <amplify-auth0-button auth0={auth0} />
          </div>
        )}

        <amplify-strike>{I18n.get('or')}</amplify-strike>
      </div>
    );
  }
}
