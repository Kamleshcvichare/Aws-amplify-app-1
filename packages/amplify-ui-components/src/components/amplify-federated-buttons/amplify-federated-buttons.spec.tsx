import { newSpecPage } from '@stencil/core/testing';

import { AmplifyFederatedButtons } from './amplify-federated-buttons';
import * as stories from './amplify-federated-buttons.stories';

const {
  default: { title },
  ...specs
} = stories;

const components = [AmplifyFederatedButtons];

describe(title, () => {
  describe('stories', () => {
    Object.entries(specs).forEach(([name, spec]) => {
      it(name, async () => {
        const page = await newSpecPage({ components, html: spec() });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
