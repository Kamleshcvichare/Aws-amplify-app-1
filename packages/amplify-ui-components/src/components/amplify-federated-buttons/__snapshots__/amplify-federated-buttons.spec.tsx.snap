// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-federated-buttons stories renderNothingWithIncorrectAuthState 1`] = `null`;

exports[`amplify-federated-buttons stories renderNothingWithNoFederatedProp 1`] = `null`;

exports[`amplify-federated-buttons stories renderWithCorrectAuthState 1`] = `null`;

exports[`amplify-federated-buttons stories renderWithCorrectAuthStateAndOnlyFacebookId 1`] = `null`;

exports[`amplify-federated-buttons stories renderWithCorrectAuthStateAndOnlyGoogleId 1`] = `null`;
